airbyte-protocol/protocol-models/src/main/java/io/airbyte/protocol/models/CatalogHelpers.java

```diff
@@ -221,15 +231,81 @@ protected static List<Pair<List<String>, JsonNode>> getFullyQualifiedFieldNamesW
     // however.
     final Set<List<String>> fieldNamesThatAreOneOfs = new HashSet<>();
 
-    return JsonSchemas.traverseJsonSchemaWithCollector(jsonSchema, (node, basicPath) -> {
+    final List<Pair<List<String>, JsonNode>> res =  JsonSchemas.traverseJsonSchemaWithCollector(jsonSchema, (node, basicPath) -> {
       final List<String> fieldName = basicPath.stream().filter(fieldOrList -> !fieldOrList.isList()).map(FieldNameOrList::getFieldName).toList();
       return Pair.of(fieldName, node);
     })
         .stream()
         // first node is the original object.
         .skip(1)
-        .filter(fieldWithSchema -> filterChildrenOfFoneOneOf(fieldWithSchema.getLeft(), fieldWithSchema.getRight(), fieldNamesThatAreOneOfs))
-        .toList();
+        .collect(
+            () -> new ContextWrapper(),
```

Comment:
For readability, this seems like the inputs to the `collect` method could be extracted to methods and/or constants for the parameter functions themself.  It will also make it easier to test the components in here.

Reply:
I agree, I made this PR to illustrate the potential issue with having a filter that rely on a state. It was not meant to be merge as is.